lib/pwwka/transmitter.rb

```diff
@@ -25,17 +30,38 @@ def initialize
     # routing_key:: String routing key for the message
     # delayed:: Boolean send this message later
     # delay_by:: Integer milliseconds to delay the message
+    # on_error:: What is the behavior of
+    # - :ignore (aka as send_message_safely)
+    # - :raise
+    # - :resque -- use Resque to try to send the message later
     #
     # Returns true
     #
     # Raises any exception generated by the innerworkings of this library.
-    def self.send_message!(payload, routing_key, delayed: false, delay_by: nil)
+    def self.send_message!(payload, routing_key, on_error: :raise,
+        delayed: false, delay_by: nil)
       if delayed
         new.send_delayed_message!(*[payload, routing_key, delay_by].compact)
       else
         new.send_message!(payload, routing_key)
       end
       info "AFTER Transmitting Message on #{routing_key} -> #{payload}"
+    rescue => e
+      error "ERROR Transmitting Message on #{routing_key} -> #{payload}: #{e}"
+      case on_error
+        when :raise
+          raise e
+        when :resque
+          send_message_async(payload, routing_key, delay_by_ms: delayed ? delay_by || DEFAULT_DELAY_BY_MS : 0)
+        else # ignore
+      end
+      false
+    end
+
+    # Use Resque to enqueue the message.
+    # - :delay_by_ms:: Integer milliseconds to delay the message. Default is 0.
+    def self.send_message_async(payload, routing_key, delay_by_ms: 0)
+      Resque.enqueue_in(delay_by_ms/1000, SendMessageAsyncJob, payload, routing_key)
```

Comment:
Right, just thinking that if there's a Resque error here, it masks the error that brought us to this codepath.  So, maybe something like:

```ruby
rescue => e
  case on_error
  when :resque
    begin
      send_message_async(payload, routing_key, delay_by_ms: delayed ? delay_by || DEFAULT_DELAY_BY_MS : 0)
    rescue => resque_exception
      warn(resque_exception.message)
      raise e
      # or something more involved like raise e.class,e.message + ", and got #{resque_exception} while sending async"
    end
```